Source/JavaScriptCore/wasm/WasmLLIntGenerator.cpp

```diff
@@ -1886,6 +1889,39 @@ auto LLIntGenerator::addI31GetU(ExpressionType ref, ExpressionType& result) -> P
     return { };
 }
 
+auto LLIntGenerator::addStructNew(uint32_t index, Vector<ExpressionType>& args, ExpressionType& result) -> PartialResult
+{
+    // We have to materialize the arguments here since it might include constants or
+    // delayed moves, but the wasm_struct_new opcode expects all the arguments to be contiguous
+    // in the stack. The reason we don't call materializeConstantsAndLocals here is that
```

Comment:
Can we change it to be polymorphic, or change what it expect?

Reply:
Yeah, we could make `materializeConstantsAndLocals` to be polymorphic but the main point here is that `materializeConstantsAndLocals` works with containers of `TypedExpression`s but here we have only container of `ExpressionType`s so that is why we can't use it here. The comment is misleading a bit, so I've shortened it.